storage/pc-ata/ata.c

```diff
@@ -0,0 +1,533 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Generic ATA controller driver
+ *
+ * Copyright 2012-2015, 2019, 2020 Phoenix Systems
+ * Author: Marcin Stragowski, Kamil Amanowicz, Lukasz Kosinski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <errno.h>
+#include <stdlib.h>
+#include <stdint.h>
+#include <string.h>
+#include <unistd.h>
+
+#include <sys/list.h>
+#include <sys/threads.h>
+#include <sys/types.h>
+
+#include <arch/ia32/io.h>
+
+#include "ata.h"
+
+
+ata_common_t ata_common;
+
+
+static ata_bus_t *buses;
+
+
+static uint32_t ata_readreg(void *base, uint8_t reg, uint8_t size)
+{
+	uintptr_t reg_base = (uintptr_t)base;
+
+	if (reg_base & 0x1) {
+		/* Read from IO-port */
+		reg_base &= ~0x3;
+		reg_base += reg;
+
+		switch (size) {
+		case 1:
+			return inb((void *)(reg_base));

```

Comment:
No need for putting reg_base in brackets. This applies to the whole commit.

Reply:
Fixed.